### PR TITLE
6426678: (spec) File.createTempFile(prefix, suffix, dir) needs clarification for illegal symbols in suffix

### PR DESCRIPTION
**Pull request by bplb**
Add some verbiage indicating that an `IOException` will be thrown if a file with a name generated from the supplied prefix and suffix according to the described algorithm cannot be generated by the underlying system, whether that be due to the presence of one of more characters not supported by the underlying system or for some other reason.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8338009](https://bugs.openjdk.org/browse/JDK-8338009) to be approved

### Issues
 * [JDK-6426678](https://bugs.openjdk.org/browse/JDK-6426678): (spec) File.createTempFile(prefix, suffix, dir) needs clarification for illegal symbols in suffix (**Bug** - P4)
 * [JDK-8338009](https://bugs.openjdk.org/browse/JDK-8338009): (spec) File.createTempFile(prefix, suffix, dir) needs clarification for illegal symbols in suffix (**CSR**)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20435/head:pull/20435` \
`$ git checkout pull/20435`

Update a local copy of the PR: \
`$ git checkout pull/20435` \
`$ git pull https://git.openjdk.org/jdk.git pull/20435/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20435`

View PR using the GUI difftool: \
`$ git pr show -t 20435`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20435.diff">https://git.openjdk.org/jdk/pull/20435.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20435#issuecomment-2264267896)

### CODE REVIEW

**Comment by bplb** [comment]
/csr to be created when verbiage changes are quiescent.

File: src/java.base/share/classes/java/io/File.java [hunk 1]
```diff
@@ -2117,6 +2117,11 @@ static File generateFile(String prefix, String suffix, File dir)
      * made the name of the new file will be generated by concatenating the
      * prefix, five or more internally-generated characters, and the suffix.
      *
+     * <p> If a file having the generated name cannot be created by the
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/io/File.java:2120]: I think it might be clear to say "with the generated name".

Comment (bplb) [reply src/java.base/share/classes/java/io/File.java]: So changed in 64440ec.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**